app/models/customer.rb

```diff
@@ -0,0 +1,8 @@
+class Customer < ApplicationRecord
+  has_many :rentals
+  has_many :movies, through: :rentals
+
+  validates :name, presence: true
+  validates :phone, presence: true
+  validates :phone, length: { is: 14 }
```

Comment:
You might be over-validating here. Your app probably won't break if the customer's phone number isn't right, so validating it (and testing the validation, and making your tests provide the phone number in the right format) is more trouble than it's worth.

Reply:
Same thing goes for many of your model validations.